admin/Openedx_Woocommerce_Plugin_Admin.php

```diff
@@ -105,85 +110,138 @@ public function enqueue_styles() {
+	{
 
 		$this->openedx_enrollment->register_enrollment_custom_post_type();
-
 	}
 
 	/**
 	 * Render Enrollment Request info form
 	 *
 	 * @since    1.0.0
 	 */
-	public function render_enrollment_info_form($post) {
+	public function render_enrollment_info_form($post)
+	{
 
 		$this->openedx_enrollment_info_form = new Openedx_Woocommerce_Plugin_Enrollment_Info_Form($post);
+	}
+
+	/**
+	 * Wrapper function to register a new post type
+	 *
+	 * @param  string $post_type   Post type name.
+	 * @param  string $plural      Post type item plural name.
+	 * @param  string $single      Post type item single name.
+	 * @param  string $description Description of post type.
+	 * @return object              Post type class object
+	 */
+	public function register_post_type($post_type = '', $plural = '', $single = '', $description = '', $options = array())
+	{
+
+		if (!$post_type || !$plural || !$single) {
+			return;
+		}
+
+		$post_type = $this->createPostType($post_type, $plural, $single, $description, $options);
+
+		return $post_type;
+	}
+
+	/**
+	 * Create a new instance of the Openedx_Woocommerce_Plugin_Post_Type class and register a new post type.
+	 *
+	 * @param  string $post_type   Post type name.
+	 * @param  string $plural      Post type item plural name.
+	 * @param  string $single      Post type item single name.
+	 * @param  string $description Description of the post type.
+	 * @param  array  $options     Additional options for the post type.
+	 * @return object              Post type class object.
+	 */
+	public function createPostType($post_type = '', $plural = '', $single = '', $description = '', $options = array())
+	{
+
+		return new Openedx_Woocommerce_Plugin_Post_Type($post_type, $plural, $single, $description, $options);
+	}
 
+	/**
+	 * Register course ID and mode fields for product
+	 *
+	 * @since    1.1.1
+	 */
+	function add_custom_product_fields()
+	{
+		global $post;
+
+		echo '<div class="options_group">';
+
+		echo '<p class="form-field">' . __('Only use these fields if the product is an Open edX course.', 'woocommerce') . '</p>';
+
+		woocommerce_wp_text_input(array(
+			'id'          => '_course_id',
+			'label'       => __('Course ID', 'woocommerce'),
+			'placeholder' => '',
+			'desc_tip'    => 'true',
+			'description' => __('Enter the course ID for the product.', 'woocommerce')
+		));
+
+		woocommerce_wp_select(array(
+			'id'          => '_mode',
+			'label'       => __('Mode', 'woocommerce'),
+			'desc_tip'    => 'true',
+			'description' => __('Select the mode for the product.', 'woocommerce'),
+			'options'     => array(
+				'Honor'             => __('Honor', 'woocommerce'),
+				'Audit'             => __('Audit', 'woocommerce'),
+				'Verified'          => __('Verified', 'woocommerce'),
+				'Credit'            => __('Credit', 'woocommerce'),
+				'Professional'      => __('Professional', 'woocommerce'),
+				'No ID Professional' => __('No ID Professional', 'woocommerce'),
+			)
```

Comment:
I think this is the second place you are using this list of course modes. Maybe it's a good idea to put this array in another file, like utils or something like that, and reuse that array instead of having both.

Reply:
Can we use those utils in our Enrollment Request object form?

Reply:
Apologies for the delay. I was mistaken and thought I had already uploaded the change. I will be making the necessary commit to implement the change shortly.